libraries/lib-uuid/Uuid.h

```diff
@@ -0,0 +1,51 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file Uuid.h
+ @brief Declare a class to generate and parse UUIDs
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <string>
+
+namespace audacity
+{
+
+class UUID_API Uuid final
+{
+public:
+   using Bytes = std::array<uint8_t, 16>;
+
+   Uuid();
+   explicit Uuid(const Bytes& data) noexcept;
+
+   static Uuid Generate();
+   static Uuid FromString(const std::string& str);
+
+   bool isNil() const noexcept;
+
+   bool operator==(const Uuid& other) const noexcept;
+   bool operator!=(const Uuid& other) const noexcept;
+
+   bool operator>(const Uuid& other) const noexcept;
```

Comment:
Comment that these are lexicographical?

Reply:
Is one purpose of this to allow a set or map of these values?

Is it worth the trouble also to customize std::hash<Uuid> for unordered containers too?

Reply:
Excellent point, I will implement it